diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/chapter3/CollapseIfStatementsRule.kt

```diff
@@ -0,0 +1,31 @@
+package org.cqfn.diktat.ruleset.rules.chapter3
+
+import org.cqfn.diktat.common.config.rules.RulesConfig
+import org.cqfn.diktat.ruleset.constants.Warnings
+import org.cqfn.diktat.ruleset.rules.DiktatRule
+import org.cqfn.diktat.ruleset.utils.findChildrenMatching
+
+import com.pinterest.ktlint.core.ast.ElementType.IF
+import org.jetbrains.kotlin.com.intellij.lang.ASTNode
+import org.jetbrains.kotlin.psi.KtIfExpression
+
+class CollapseIfStatementsRule(configRules: List<RulesConfig>) : DiktatRule(
+    "collapse-if",
+    configRules,
+    listOf(
+        Warnings.COLLAPSE_IF_STATEMENTS
+    )
+) {
+    override fun logic(node: ASTNode) {
+        if (node.elementType == IF) {
+            val thenNode = (node.psi as KtIfExpression).then?.node
+            val nestedIf = thenNode?.findChildrenMatching(IF) { true }?.firstOrNull()
```

Comment:
@petertrr I was not able to find quickly, but I remember that we had a method that finds ChildByElementType

Reply:
`findAllDescendantsWithSpecificType`?

Reply:
Done in [code line](https://github.com/cqfn/diKTat/pull/814/commits/a6fffa24a2675831884963fd04fc44e710eed820#diff-251ce198ca409fe76ce1e2c0c723f67bf4f143f4351c163c1a86175f5be5f397R33)